kratos/geometries/line_3d_3.h

```diff
@@ -605,6 +607,91 @@ class Line3D3 : public Geometry<TPointType>
         return rResult;
     }
 
+    /**
+     * @brief Returns the local coordinates of a given arbitrary point
+     * @param rResult The vector containing the local coordinates of the point
+     * @param rPoint The point in global coordinates
+     * @return The vector containing the local coordinates of the point
+     */
+    CoordinatesArrayType& PointLocalCoordinates(
+            CoordinatesArrayType& rResult,
+            const CoordinatesArrayType& rPoint
+            ) const override
+    {
+        // Define distance_objective as the gradient of ||point - global_coordinate(xi)||**2,
+        // that is (point - local_coordinate(xi))*coordinate_derivative(xi) (times -2, which we ignore)
+        // The zeros of the distance_objective are local extremes of the distance between point and line
+        // We will get the zeros contained in the interval [-1, 1] (if any). If, for any of these zeros,
+        // the point is on the line, the local coordinate is our result.
+        rResult.clear();
+
+        constexpr double TOLERANCE = 1e-12;
+
+        const auto& rP0 = this->GetPoint(0);
```

Comment:
Maybe r_p0 instead

Reply:
Done

side rant: I'm starting to hate the typing conventions for variable names (r..., p...). This one here is particularly useless, since a const ref is functionally identical to a const value 🤷